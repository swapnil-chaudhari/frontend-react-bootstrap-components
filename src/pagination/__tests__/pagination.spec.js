import React from 'react';
import renderShallow from 'render-shallow';
import noop from 'src/utils/noop';
import Pagination from '../pagination';

describe('<Pagination>', () => {
    describe('when it renders', () => {
        const props = {
            items: 20,
            maxButtons: 5,
            activePage: 2,
            onSelect: noop
        };

        let component;

        beforeAll(() => {
            component = renderShallow(<Pagination { ...props } />).output;
        });

        test('should have proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with size attribute', () => {
        const props = {
            items: 20,
            maxButtons: 5,
            activePage: 2,
            onSelect: noop,
            size: 'small'
        };

        let component;

        beforeAll(() => {
            component = renderShallow(<Pagination { ...props } />).output;
        });

        test('adds classname Pagination-small for component wrapper', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
