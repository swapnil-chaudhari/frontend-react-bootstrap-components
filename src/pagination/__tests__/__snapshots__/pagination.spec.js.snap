// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination> when it renders should have proper markup 1`] = `
<div
  className="Pagination"
>
  <Pagination
    activePage={2}
    boundaryLinks={true}
    bsClass="pagination"
    buttonComponentClass={[Function]}
    ellipsis={true}
    first={false}
    items={20}
    last={false}
    maxButtons={5}
    next={<RightPaginationArrow />}
    onSelect={[Function]}
    prev={<LeftPaginationArrow />}
  />
</div>
`;

exports[`<Pagination> when it renders with size attribute adds classname Pagination-small for component wrapper 1`] = `
<div
  className="Pagination Pagination-small"
>
  <Pagination
    activePage={2}
    boundaryLinks={true}
    bsClass="pagination"
    buttonComponentClass={[Function]}
    ellipsis={true}
    first={false}
    items={20}
    last={false}
    maxButtons={5}
    next={<RightPaginationArrow />}
    onSelect={[Function]}
    prev={<LeftPaginationArrow />}
  />
</div>
`;
